Launcher.py

```diff
@@ -107,7 +108,55 @@ def update_settings():
 ])
 
 
-def identify(path: Union[None, str]):
+def handle_uri(path: str, launch_args: Tuple[str, ...]) -> None:
+    url = urllib.parse.urlparse(path)
+    queries = urllib.parse.parse_qs(url.query)
+    client_component = None
+    text_client_component = None
+    if "game" in queries:
+        game = queries["game"][0]
+    else:  # TODO around 0.5.0 - this is for pre this change webhost uri's
+        game = "Archipelago"
+    for component in components:
+        if component.supports_uri and component.game_name == game:
+            client_component = component
+        elif component.display_name == "Text Client":
+            text_client_component = component
+
+    from kvui import App, Button, BoxLayout, Label
```

Comment:
```suggestion
    from kvui import App, Button, BoxLayout, Label
    
    if client_component is None:
        run_component(text_client_component, *launch_args)
        return
```

I would much prefer this for ux.
However it seems like opinion is split on this.
Maybe we can poll it?

Reply:
(This would also mean that the later check for `if client_component is not None:` could be removed)

Reply:
Alternative we discussed on Discord: A little timer window that says "No client detected, launching Text Client in 5s" or something